### PR TITLE
KOGITO-2431: Fixed snapshots related to DE

### PR DESCRIPTION
**Pull request by Sara4994**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: packages/common/src/components/Molecules/DomainExplorerTable/tests/DomainExplorerTable.test.tsx [hunk 1]
```diff
@@ -211,14 +216,17 @@ describe('Domain Explorer Table Component', () => {
...
     };
 
-    const wrapper = shallow(<DomainExplorerTable {...props} />);
-
+    const wrapper = await getWrapperAsync(
+      <MockedProvider>
+        <DomainExplorerTable {...props} />
+      </MockedProvider>,
+      'DomainExplorerTable'
+    );
```
Comment (jstastny-cz) [inline packages/common/src/components/Molecules/DomainExplorerTable/tests/DomainExplorerTable.test.tsx]: I think the following wrapper.update() is not needed when using the wrappers.

File: packages/common/src/components/Molecules/DomainExplorerTable/tests/DomainExplorerTable.test.tsx [hunk 2]
```diff
@@ -15,7 +16,7 @@ describe('Domain Explorer Table Component', () => {
     mockUseEffect();
     mockUseEffect();
   });
-  it('Snapshot test', () => {
+  it('Snapshot test with default props', async () => {
```
Comment (jstastny-cz) [inline packages/common/src/components/Molecules/DomainExplorerTable/tests/DomainExplorerTable.test.tsx]: The resulting snapshot is huge, can we limit the size of it by selective mocking?

Probably the Table component spans the most lines, though I believe it is important to keep at least one snapshot showing also the rendered Table component - can we identify a few tests that actually show different states of the rendered Table? And keep full snapshots there.
And mock the Table in the others - where it is not important (like the test for isLoading:false and similar, there the full-table snapshot might not be needed).

File: packages/common/src/components/Organisms/DomainExplorer/tests/__snapshots__/DomainExplorer.test.tsx.snap [hunk 3]
```diff
@@ -1505,6 +873,942 @@ exports[`Domain Explorer Container component Snapshot test 1`] = `
...
+                    </Spinner>
+                  </div>
+                </EmptyStateIcon>
+                <Title
+                  size="lg"
+                >
+                  <h1
+                    className="pf-c-title pf-m-lg"
+                  >
+                    Loading domain data...
```
Comment (jstastny-cz) [inline packages/common/src/components/Organisms/DomainExplorer/tests/__snapshots__/DomainExplorer.test.tsx.snap]: Here the data is still loading - try getWrapperAsync + async test.

File: packages/common/src/components/Organisms/DomainExplorer/tests/__snapshots__/DomainExplorer.test.tsx.snap [hunk 4]
```diff
@@ -1505,6 +873,942 @@ exports[`Domain Explorer Container component Snapshot test 1`] = `
...
```
Comment (jstastny-cz) [inline packages/common/src/components/Organisms/DomainExplorer/tests/__snapshots__/DomainExplorer.test.tsx.snap]: data still loading - try getWrapperAsync + async test

File: packages/common/src/components/Organisms/DomainExplorer/tests/__snapshots__/DomainExplorer.test.tsx.snap [hunk 5]
```diff
@@ -1505,6 +873,942 @@ exports[`Domain Explorer Container component Snapshot test 1`] = `
...
+      Object {
+        "flight": Array [
+          "departure",
+        ],
+      },
+    ]
+  }
+  rememberedSelections={Array []}
+>
+  <DataToolbar
```
Comment (jstastny-cz) [inline packages/common/src/components/Organisms/DomainExplorer/tests/__snapshots__/DomainExplorer.test.tsx.snap]: I think we can mock DataToolbar, right? There is no behavior change expected in various tests.

File: packages/common/src/components/Organisms/DomainExplorerListDomains/tests/__snapshots__/DomainExplorerListDomains.test.tsx.snap [hunk 6]
```diff
@@ -1,260 +1,355 @@
...
-    <DomainExplorerListDomains>
-      <EmptyState
-        variant="full"
+    <div
+      className="pf-c-empty-state"
+    >
+      <EmptyStateIcon
+        icon={[Function]}
       >
+        <CubesIcon
```
Comment (jstastny-cz) [inline packages/common/src/components/Organisms/DomainExplorerListDomains/tests/__snapshots__/DomainExplorerListDomains.test.tsx.snap]: Please mock the icon.

File: packages/management-console/src/components/Templates/DomainExplorerLandingPage/tests/__snapshots__/DomainExplorerLandingPage.test.tsx.snap [hunk 7]
```diff
@@ -1,242 +1,220 @@
...
+                          Home
+                        </a>
+                      </LinkAnchor>
+                    </Link>
+                    <span
+                      className="pf-c-breadcrumb__item-divider"
                     >
-                      <li
-                        className="pf-c-breadcrumb__item"
+                      <AngleRightIcon
```
Comment (jstastny-cz) [inline packages/management-console/src/components/Templates/DomainExplorerLandingPage/tests/__snapshots__/DomainExplorerLandingPage.test.tsx.snap]: please mock the icon

File: packages/management-console/src/components/Templates/DomainExplorerLandingPage/tests/__snapshots__/DomainExplorerLandingPage.test.tsx.snap [hunk 8]
```diff
@@ -1,242 +1,220 @@
...
-                  <DomainExplorerListDomains>
-                    <EmptyState
-                      variant="full"
+                  <div
+                    className="pf-c-empty-state"
+                  >
+                    <EmptyStateIcon
+                      icon={[Function]}
                     >
+                      <CubesIcon
```
Comment (jstastny-cz) [inline packages/management-console/src/components/Templates/DomainExplorerLandingPage/tests/__snapshots__/DomainExplorerLandingPage.test.tsx.snap]: please mock the icon

File: packages/management-console/src/components/Templates/DomainExplorerPage/tests/__snapshots__/DomainExplorerPage.test.tsx.snap [hunk 9]
```diff
@@ -98,525 +98,413 @@ exports[`Domain Explorer Dashboard component Check error response for getQueryFi
...
+                          Home
+                        </a>
+                      </LinkAnchor>
+                    </Link>
+                    <span
+                      className="pf-c-breadcrumb__item-divider"
                     >
-                      <li
-                        className="pf-c-breadcrumb__item"
+                      <AngleRightIcon
```
Comment (jstastny-cz) [inline packages/management-console/src/components/Templates/DomainExplorerPage/tests/__snapshots__/DomainExplorerPage.test.tsx.snap]: please mock icons - I think we can mock whole package.

File: packages/management-console/src/components/Templates/DomainExplorerPage/tests/__snapshots__/DomainExplorerPage.test.tsx.snap [hunk 10]
```diff
@@ -98,525 +98,413 @@ exports[`Domain Explorer Dashboard component Check error response for getQueryFi
...
+          </Breadcrumb>
+        </ComponentWithOuia>
+      </Component>
+    </section>
+  </PageSection>
+  <PageSection>
+    <section
+      className="pf-c-page__main-section"
+    >
+      <DomainExplorer
```
Comment (jstastny-cz) [inline packages/management-console/src/components/Templates/DomainExplorerPage/tests/__snapshots__/DomainExplorerPage.test.tsx.snap]: Again, huge snapshots - can we mock whole DomainExplorer? I mean we just want to check the passed properties, the rendering of DomainExplorer itself should be in the tests for that component.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by Sara4994** [comment]
opened a new https://github.com/kiegroup/kogito-apps/pull/284 due to some conflicts

**Pull request closed without merging**